### PR TITLE
Add quic support and fix some bugs

### PR DESCRIPTION
**Pull request by wj8400684**

### CODE REVIEW

File: src/SuperSocket.Connection/PipeConnectionBase.cs [hunk 1]
```diff
@@ -97,7 +97,8 @@ private async ValueTask HandleClosing()
         {
             try
             {
-                await _pipeTask.ConfigureAwait(false);
+                if (_pipeTask != null)
```
Comment (kerryjiang) [inline src/SuperSocket.Connection/PipeConnectionBase.cs:100]: add comment for the scenario _pipeTask = null

Comment (kerryjiang) [reply src/SuperSocket.Connection/PipeConnectionBase.cs:100]: You can put this change into a separated PR.

Comment (wj8400684) [reply src/SuperSocket.Connection/PipeConnectionBase.cs:100]: ok

File: src/SuperSocket.Quic.Connection/QuicPipeStream.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
+using System;
+using System.IO;
+using System.Net.Quic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace SuperSocket.Quic.Connection;
+
+#pragma warning disable CA2252
```
Comment (kerryjiang) [inline src/SuperSocket.Quic.Connection/QuicPipeStream.cs:9]: ignore ca1415 as well?

File: src/SuperSocket.Quic.Connection/QuicPipeConnection.cs [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+#pragma warning disable CA2252
+
+namespace SuperSocket.Quic.Connection;
+
+public class QuicPipeConnection : StreamPipeConnection
+{
+    private readonly Stream _stream;
+
+    public QuicPipeConnection(Stream stream, EndPoint remoteEndPoint, ConnectionOptions options)
```
Comment (kerryjiang) [inline src/SuperSocket.Quic.Connection/QuicPipeConnection.cs:18]: Does it really accept all kinds of Stream or only support QuicStream?

File: src/SuperSocket.Quic.Connection/QuicPipeStream.cs [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+    public override long Position
+    {
+        get => _stream.Position;
+        set => _stream.Position = value;
+    }
+
+    public async ValueTask OpenStreamAsync(CancellationToken cancellationToken)
+    {
+        _stream = await _connection.AcceptInboundStreamAsync(cancellationToken);
```
Comment (kerryjiang) [inline src/SuperSocket.Quic.Connection/QuicPipeStream.cs:35]: It's inbound stream, do we need handle the outbound stream?

Comment (wj8400684) [reply src/SuperSocket.Quic.Connection/QuicPipeStream.cs:35]: The Inbound Stream seems to be all that's needed on the server

File: src/SuperSocket.Quic.Connection/QuicPipeStream.cs (hunk 2 above)
Comment (wj8400684) [reply src/SuperSocket.Quic.Connection/QuicPipeStream.cs:9]: image

**Pull request closed without merging**